### DISCUSSION TITLE
DEV SESSION 16 DEC --- 2000 UTC+8/1300 CET/1200 GMT/ 0700 EST

### DISCUSSION THREAD

**Discussion by jackiekazil** [mesa/mesa#1897] [General]
Where: [via Matrix](https://matrix.to/#/#project-mesa:matrix.org)

Proposed Agenda:

- General updates 
- Google Summer of Code 2024 & NumFOCUS
- Creating MEP? (Mesa Enhancement Proposal) - see https://github.com/projectmesa/mesa/issues/1889#issuecomment-1838804212 
- Front-end 
- #1894 (for me the highest priority)
- #1898
---------
- #1625
- #348 (including properly addressing #1701)
- [Mesa interactive](https://github.com/Corvince/mesa-interactive) integration
- #1717 (and #1625)
- #1756
- #1900
- Other PRs & Issues

@rht, @tpike3 , @Corvince , @wang-boyu , @ankitk50, @EwoutH

**Comment by jackiekazil** [comment]
I used a time zone converter. Can you all do a check and let me know if I have it off?

**Reply by EwoutH** [reply]
Thanks for the invite, CET looks right!

**Reply by wang-boyu** [reply]
Yup EST is correct. Thanks Jackie.

**Comment by EwoutH** [comment]
I can't edit the agenda, but I would like to add:
- #1894
   (for me the highest priority)
- #1625
- #348 (including properly addressing #1701)
- [Mesa interactive](https://github.com/Corvince/mesa-interactive) integration
- #1717 (and #1625)
- #1756

**Reply by jackiekazil** [reply]
I added them.

**Reply by EwoutH** [reply]
Two more! (I hope it fits)
- Of course the big one: https://github.com/projectmesa/mesa/pull/1898
- And a little one: https://github.com/projectmesa/mesa/issues/1900

**Reply by EwoutH** [reply]
@jackiekazil I wrote another two proposals I would like to discuss:

- #1903
- #1905

To speed development up, maybe if we can agree on the proposal conceptually and on the API during the dev meeting, and then a PR only needs to be reviewed on implementation.

(since policy cc @tpike3)

**Comment by EwoutH** [comment]
@jackiekazil Are you still thinking / working on a kind of foundation for Mesa? If you have some ideas, I could advocate them at the TU Delft side of things. We teach Mesa to over 400 students yearly in at least six different courses spread among a bachelor and two master programs, aside from researchers using it. So there is interest, and at this point everyone knows that tooling is currently the limiting factor for many ABM projects (either that or data).

Personally, I'm thinking about some kind of cooperation where every participant contributes engineering hours and/or a monetary contribution. If you can set up that with 3 or 4 universities and get a paid maintainer, even just for something like 0.2 FTE, it will really accelerate Mesa development (see what I have done in just under 2 weeks part time, as reference). Furthermore, we could dedicate bounties or budgets to certain desired features.

**Reply by jackiekazil** [reply]
yes... I have done some work on this front. I think you propose an interesting idea though. Let's set up a separate discussion for this to focus this meeting on the main items in flight now. 

But I do like this... I will reach out.

**Reply by wang-boyu** [reply]
Not to deviate from the meeting... I recall mentioning a company named Posit a few meetings ago, a company that was previously known as RStudio and is maintaining many popular open-source R packages for data science. Thought that it might be interesting to know more about them if you'd like to pursue a similar path forward. Recently the creator of Pandas has also joined the company: https://posit.co/blog/welcome-wes/

**Reply by EwoutH** [reply]
Ideally we would have a mix of academic and corporate sponsors.

**Comment by EwoutH** [comment]
I would like to put this idea on the agenda, preferable somewhere high.

- https://github.com/projectmesa/mesa/discussions/1909

**Comment by jackiekazil** [comment]
# Summary of notes from meeting 

**Google summer of code** 
We need to find another mentor. We prefer to have two for support. Jackie is one. 
Todo: Jackie will ask on to see if others can mentor 

**MEPs -**
What should be MEP?
When is a MEP required? 
Do we need to talk about implementation users?
Todo: Writing the first MEP -- how do they work (Jackie) 

**Ankit FE updates** 
1902 - Ankit integrated alteria grid
https://github.com/projectmesa/mesa/pull/1902 
3 tests -- browser, colab, jupyter lab
Added tooling for agent-based statistics
Rebecca has done with success but found a bug -- https://simrisk.pulcloud.io/riskyfood
TODO: request - Ankit will take a look at Rebecca’s PR
We should be to do another release this week
Meetings -- Dec 22, Jan 5... ends on Feb 10 
Tom will meet with him 

**Native support for multiple agent types** 
https://github.com/projectmesa/mesa/pull/1894 
Right now you have to go through the scheduler -- this is so you can divorce it. 
Model contains everything. 
Agents aren’t connected to the model in any way.
With scheduler - self.scheduler; with space self.space; with agent self.agent 
Todos: 
* Tom and Jackie will do a review. 
* Jackie will provide Tom with the original diagram
* Tom will update the diagram 
* Ewout will write tests
* Ewout will update the Bolzman tutorial

**Experimental features** 
https://github.com/projectmesa/mesa/discussions/1909
They don’t have follow semantic versioning
They don’t have to fully be reviewed 
We clearly identify that as a change
“If we write something on Tuesday, then we can test on Wednesday”
ToDos: 
Everyone review / respond 
When Jackie drafts the MEP, ping Ewout to review/consider

**Tom will work on history for 2.2.0 (Tom)**

**Releases -- consider automation?**
Ewout does this - https://github.com/quaquel/EMAworkbench/blob/master/CONTRIBUTING.md#releases 
Should we automate?
Todo: Review and come up with it.
Todo: Setup time with Ewout to go over release process (Jackie & Ewout)
We should have a release branch
Each minor release gets it own release branch 
Related: https://github.com/projectmesa/mesa/pull/1479
